pkg/cache/cache_test.go

```diff
@@ -3378,3 +3378,310 @@ func TestClusterQueueReadiness(t *testing.T) {
+		}
+	})
+
+	t.Run("clusterqueue joining cohort with cycle successfully updates old cohort", func(t *testing.T) {
+		cache := New(utiltesting.NewFakeClient())
+		ctx := context.Background()
+		cycleCohort := utiltesting.MakeCohort("cycle").Parent("cycle").Obj()
+		if err := cache.AddCohort(cycleCohort); err == nil {
+			t.Fatal("Expected failure")
+		}
+		cohort := utiltesting.MakeCohort("cohort").
+			ResourceGroup(utiltesting.MakeFlavorQuotas("arm").Resource(corev1.ResourceCPU, "10").FlavorQuotas).Obj()
+		if err := cache.AddCohort(cohort); err != nil {
+			t.Fatal("Expected success")
+		}
+
+		// Add CQ to cohort
+		cq := utiltesting.MakeClusterQueue("cq").
+			ResourceGroup(utiltesting.MakeFlavorQuotas("arm").Resource(corev1.ResourceCPU, "5").FlavorQuotas).
+			Cohort("cohort").Obj()
+		if err := cache.AddClusterQueue(ctx, cq); err != nil {
+			t.Fatal("Expected success")
+		}
+
+		// cohort's SubtreeQuota contains resources from cq
+		gotResource := cache.hm.Cohorts["cohort"].getResourceNode()
+		wantResource := ResourceNode{
+			Quotas: map[resources.FlavorResource]ResourceQuota{
+				{Flavor: "arm", Resource: corev1.ResourceCPU}: {Nominal: 10_000},
+			},
+			SubtreeQuota: resources.FlavorResourceQuantities{
+				{Flavor: "arm", Resource: corev1.ResourceCPU}: 15_000,
+			},
+			Usage: resources.FlavorResourceQuantities{},
+		}
+		if diff := cmp.Diff(wantResource, gotResource); diff != "" {
+			t.Errorf("Unexpected resource (-want,+got):\n%s", diff)
+		}
+
+		// Updated to cycle
+		cq.Spec.Cohort = "cycle"
+		if err := cache.UpdateClusterQueue(cq); err == nil {
+			t.Fatal("Expected failure")
+		}
+
+		// Cohort's SubtreeQuota no longer contains resources from CQ.
+		gotResource = cache.hm.Cohorts["cohort"].getResourceNode()
+		wantResource = ResourceNode{
+			Quotas: map[resources.FlavorResource]ResourceQuota{
+				{Flavor: "arm", Resource: corev1.ResourceCPU}: {Nominal: 10_000},
+			},
+			SubtreeQuota: resources.FlavorResourceQuantities{
+				{Flavor: "arm", Resource: corev1.ResourceCPU}: 10_000,
+			},
+			Usage: resources.FlavorResourceQuantities{},
+		}
+		if diff := cmp.Diff(wantResource, gotResource); diff != "" {
+			t.Errorf("Unexpected resource (-want,+got):\n%s", diff)
+		}
+	})
+
+	t.Run("cohort switching cohorts updates both cohort trees", func(t *testing.T) {
+		cache := New(utiltesting.NewFakeClient())
+		root1 := utiltesting.MakeCohort("root1").Obj()
+		root2 := utiltesting.MakeCohort("root2").Obj()
+		if err := cache.AddCohort(root1); err != nil {
+			t.Fatal("Expected success")
+		}
+		if err := cache.AddCohort(root2); err != nil {
+			t.Fatal("Expected success")
+		}
+
+		cohort := utiltesting.MakeCohort("cohort").Parent("root1").
+			ResourceGroup(utiltesting.MakeFlavorQuotas("arm").Resource(corev1.ResourceCPU, "10").FlavorQuotas).Obj()
+		if err := cache.AddCohort(cohort); err != nil {
+			t.Fatal("Expected success")
+		}
+
+		t.Run("before move", func(t *testing.T) {
```

Comment:
do not use `t.Run` for non-independent tests cases

Reply:
Done.